mslice/models/slice/mantid_slice_algorithm.py

```diff
@@ -65,15 +66,24 @@ def compute_chi_magnetic(self, chi):
         chi_magnetic = chi / 291
         return chi_magnetic
 
+    def compute_d2sigma(self, scattering_data, workspace, y_axis):
+        Ei = self._workspace_provider.get_EFixed(self._workspace_provider.get_parent_by_name(workspace))
+        if Ei is None:
+            return None
+        ki = np.sqrt(Ei*2*constants.neutron_mass) / HBAR_MEV
+        energy_transfer = np.arange(y_axis.end, y_axis.start, -y_axis.step)
+        kf = np.sqrt(((Ei - energy_transfer)*2*constants.neutron_mass) / HBAR_MEV)[:, None]
```

Comment:
I'm not sure this is right. I don't think the `sqrt` should encompass the `hbar`? I suspect this is the reason the values are so small. I would expect that `kf/ki` would be `1` `deltaE=0` and then decrease as `deltaE` increases.

Further to this I think we might be as well to define a conversion constant for `E`->`K` at module level so that we do that calculation once, i.e.

```
E_TO_K = np.sqrt(2*constants.neutron_mass)/HBAR_MEV
```

Then

```
ki = np.sqrt(Ei)*E_TO_K
kf = np.sqrt(Ei - deltaE)*E_TO_K
etc
```